src/threading/MsgThread.cc

```diff
@@ -178,6 +179,29 @@ MsgThread::MsgThread() : BasicThread(), queue_in(this, 0), queue_out(0, this)
 	child_finished = false;
 	failed = false;
 	thread_mgr->AddMsgThread(this);
+
+	// Open a socket pair used to trigger the iosource loop into processing
+	// new messages on this thread. It's set to non-block because we want to
+	// read out all notifications on every pass without blocking.
+	socketpair(AF_UNIX, SOCK_STREAM, 0, ready_pair);
```

Comment:
Probably doesn't matter much, but likely can get away with `pipe` instead of `socketpair` since only need to write bytes going in one direction.  Also this seems like the kind of thing a `Flare` object from the `Flare.cc` code could be re-used for if you find that useful/easier ?  Just mentioning it because I think that already would address a few of the other nitpicks I commented about.

Reply:
I tried `pipe()` originally and for some reason I couldn't ever get it to work. The poll would never wake up for the writes. I'll revisit it.

Reply:
After looking at the `Flare` code, I agree that would be better since I wouldn't need to manage it myself. I'll switch over to that.